src/main/scala/com/microsoft/hyperspace/index/IndexManager.scala

```diff
@@ -104,4 +104,6 @@ trait IndexManager {
    * @return Index metadata and statistics as a [[DataFrame]].
    */
   def index(indexName: String): DataFrame
+
+  private[hyperspace] def getLogManager(indexName: String): Option[IndexLogManager]
```

Comment:
I am not sure if this is a good idea to expose the `IndexLogManager` at this point (I understand this is package private, but since this can be used from source relation APIs, we should treat this as a public API).

One way is to make `IndexManager` to be more of a facade to the `IndexLogManager` and provide `def getIndexVersions(name: String): Option[Seq[Int]]` and `def getIndex(name: String, version: Int): Option[IndexLogEntry]`. `CachingIndexCollectionManager` can be used to cache `IndexLogEntry` this way if needed.

Any thoughts?